lib-rpc/ocamlformat_rpc_lib.mli

```diff
@@ -9,70 +9,78 @@
 (*                                                                        *)
 (**************************************************************************)
 
-module type Command_S = sig
-  type t
+module Make (IO : IO.S) : sig
+  module type Command_S = sig
+    type t
 
-  val read_input : Stdlib.in_channel -> t
+    val read_input : IO.ic -> t IO.t
 
-  val to_sexp : t -> Sexplib0.Sexp.t
+    val to_sexp : t -> Csexp.t
 
-  val output : Stdlib.out_channel -> t -> unit
-end
+    val output : IO.oc -> t -> unit IO.t
+  end
 
-module type Client_S = sig
-  type t
+  module type V = sig
+    module Command : Command_S
 
-  type cmd
+    module Client : sig
+      type t
 
-  val pid : t -> int
+      type cmd
 
-  val mk : pid:int -> in_channel -> out_channel -> t
+      val pid : t -> int
 
-  val query : cmd -> t -> cmd
+      val mk : pid:int -> IO.ic -> IO.oc -> t
 
-  val halt : t -> (unit, [> `Msg of string]) result
+      val query : cmd -> t -> cmd IO.t
 
-  val config :
-    (string * string) list -> t -> (unit, [> `Msg of string]) result
+      val halt : t -> (unit, [> `Msg of string]) result IO.t
+      (** The caller must close the input and output channels after calling
+          [halt]. *)
 
-  val format : string -> t -> (string, [> `Msg of string]) result
-end
+      val config :
+        (string * string) list -> t -> (unit, [> `Msg of string]) result IO.t
 
-module type V = sig
-  module Command : Command_S
+      val format : string -> t -> (string, [> `Msg of string]) result IO.t
+    end
+    with type cmd = Command.t
+  end
 
-  module Client : Client_S with type cmd = Command.t
-end
+  (** Version used to set the protocol version *)
+  module Init :
+    Command_S with type t = [`Halt | `Unknown | `Version of string]
 
-(** Version used to set the protocol version *)
-module Init : Command_S with type t = [`Halt | `Unknown | `Version of string]
+  module V1 :
+    V
+      with type Command.t =
+        [ `Halt
+        | `Unknown
+        | `Error of string
+        | `Config of (string * string) list
+        | `Format of string ]
 
-module V1 :
-  V
-    with type Command.t =
-      [ `Halt
-      | `Unknown
-      | `Error of string
-      | `Config of (string * string) list
-      | `Format of string ]
+  type client = [`V1 of V1.Client.t]
 
-type client = [`V1 of V1.Client.t]
+  val pick_client :
+       pid:int
+    -> IO.ic
+    -> IO.oc
+    -> string list
+    -> (client, [`Msg of string]) result IO.t
+  (** [pick_client ~pid in out versions] returns the most-fitting client
+      according to a list of [versions], that is a list ordered from the most
+      to the least wished version. *)
```

Comment:
```suggestion
      to the least wished version. The given `IO.ic` and `IO.oc` values are
      referenced by the `client` value and must be kept open until `halt` is
      called. *)
```

I'm not sure about my english here but I think this is worth documenting.

Reply:
Sounds good to me